model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -896,57 +876,30 @@ def _do_diffusion_step(
+                theta_v_in=prognostic_state.theta_v,
                 geofac_div=self._interpolation_state.geofac_div,
-                z_temp=self.z_temp,
-                horizontal_start=self._cell_start_nudging,
-                horizontal_end=self._cell_end_local,
-                vertical_start=0,
-                vertical_end=self._grid.num_levels,
-                offset_provider=self._grid.connectivities,
-            )
-            log.debug("running stencils 13_14 (calculate_nabla2_for_theta): end")
-            log.debug(
-                "running stencil 15 (truly_horizontal_diffusion_nabla_of_theta_over_steep_points): start"
-            )
-            if self.config.apply_zdiffusion_t:
-                self.truly_horizontal_diffusion_nabla_of_theta_over_steep_points(
-                    mask=self._metric_state.mask_hdiff,
-                    zd_vertoffset=self._metric_state.zd_vertoffset,
-                    zd_diffcoef=self._metric_state.zd_diffcoef,
-                    geofac_n2s_c=self._interpolation_state.geofac_n2s_c,
-                    geofac_n2s_nbh=self._interpolation_state.geofac_n2s_nbh,
-                    vcoef=self._metric_state.zd_intcoef,
-                    theta_v=prognostic_state.theta_v,
-                    z_temp=self.z_temp,
-                    horizontal_start=self._cell_start_nudging,
-                    horizontal_end=self._cell_end_local,
-                    vertical_start=0,
-                    vertical_end=self._grid.num_levels,
-                    offset_provider=self._grid.connectivities,
-                )
-
-                log.debug(
-                    "running fused stencil 15 (truly_horizontal_diffusion_nabla_of_theta_over_steep_points): end"
-                )
-            log.debug("running stencil 16 (update_theta_and_exner): start")
-            self.update_theta_and_exner(
-                z_temp=self.z_temp,
+                mask=self._metric_state.mask_hdiff,
+                zd_vertoffset=self._metric_state.zd_vertoffset,
+                zd_diffcoef=self._metric_state.zd_diffcoef,
+                geofac_n2s_c=self._interpolation_state.geofac_n2s_c,
+                geofac_n2s_nbh=self._interpolation_state.geofac_n2s_nbh,
+                vcoef=self._metric_state.zd_intcoef,
                 area=self._cell_params.area,
                 theta_v=prognostic_state.theta_v,
                 exner=prognostic_state.exner,
                 rd_o_cvd=self.rd_o_cvd,
+                apply_zdiffusion_t=self.config.apply_zdiffusion_t,
                 horizontal_start=self._cell_start_nudging,
                 horizontal_end=self._cell_end_local,
                 vertical_start=0,
                 vertical_end=self._grid.num_levels,
+                offset_provider=self._grid.connectivities,
             )
-            log.debug("running stencil 16 (update_theta_and_exner): end")
+            log.debug("running stencil 13 and 16 (apply_diffusion_to_theta_and_exner): end")
```

Comment:
```suggestion
            log.debug("running stencil 13 to 16 (apply_diffusion_to_theta_and_exner): end")
```

Reply:
I think we could also drop the old stencil names, i.e.
```suggestion
            log.debug("running apply_diffusion_to_theta_and_exner: end")
```
but that should be applied to the rest of the file, and can be done in a separate PR.